allauth/socialaccount/providers/pinterest/provider.py

```diff
@@ -20,8 +20,14 @@ def get_avatar_url(self):
         return self.account.extra_data.get("profile_image")
 
     def to_str(self):
-        dflt = super(PinterestAccount, self).to_str()
-        return self.account.extra_data.get("username", dflt)
+        username = self.account.extra_data.get("username")
+        if username:
+            return username
+        first_name = self.account.extra_data.get("first_name")
+        last_name = self.account.extra_data.get("last_name")
+        if first_name or last_name:
+            return f"{first_name or ''} {last_name or ''}".strip()
+        return super().to_str()
```

Comment:
Can't we rely on the base `to_str()` here? The above logic is not Pinterest specific at all.